arbitrum-client/src/client/config.rs

```diff
@@ -0,0 +1,87 @@
+//! The configuration options for the Arbitrum client, along with
+//! convenience methods for converting them into richer types that the
+//! client works with.
+
+use std::{error::Error, fmt::Display, str::FromStr, sync::Arc};
+
+use ethers::{
+    core::k256::ecdsa::SigningKey,
+    middleware::SignerMiddleware,
+    providers::{Http, Middleware, Provider},
+    signers::{LocalWallet, Signer, Wallet},
+    types::Address,
+};
+
+use crate::abi::DarkpoolContract;
+
+pub struct ArbitrumClientConfig {
+    /// The address of the darkpool proxy contract.
+    ///
+    /// This is the main entrypoint to interaction with the darkpool.
+    pub proxy_address: String,
+    /// The address of the darkpool implementation contract.
+    ///
+    /// This is used to filter for events emitted by the darkpool.
+    pub implementation_address: String,
```

Comment:
Again let's hide the fact that this is proxied, perhaps `event_source` or similar if it's only used for events.